vscode/src/client/analyzerClient.ts

```diff
@@ -143,18 +143,22 @@ export class AnalyzerClient {
   ): Promise<[ChildProcessWithoutNullStreams, number | undefined]> {
     // TODO: Ensure serverState is starting
 
-    const serverCwd = this.kaiRuntimeDir;
+    const serverCwd = this.extContext.storageUri?.fsPath as string;
     const serverPath = this.getKaiRpcServerPath();
     const serverArgs = this.getKaiRpcServerArgs();
     const serverEnv = await this.getKaiRpcServerEnv();
 
+    if (!fs.existsSync(serverCwd)) {
```

Comment:
VS Code API is preferred - it's async and better integrated with VS Code flow

Reply:
didn't find a good replacement for testing for existence of the dir in vscode api. but I did move dir creation to the vscode api.